go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/stripe/veneur/v14
 
-go 1.18
+go 1.19
```

Comment:
I don't think we want this - this bumps the minimum version of Go.  we last bumped this for 1.18, when we started using generics.  I don't see any features of 1.19 we're using, and this would restrict the ability to use veneur from a project still on 1.18

Reply:
Understandable; I'll leave this up for future consideration then.

Reply:
However, I don't see much use of Veneur as a package imported by other projects 🤔

It seems by far and away meant to be built as an application, so then again I don't really see that as a reason to constrain Go version. I could be wrong though and haven't seen other projects importing Veneur.

Reply:
@mkjois Stripe has at least 1 private Go repo that imports veneur for packages like https://pkg.go.dev/github.com/stripe/veneur/v14@v14.2.0/trace and is still at go 1.18

I believe we should only bump go.mod if we actually start using language features in newer versions of Go - that can happen in individual PRs that make use of new features.  Is there an advantage to preemptively bumping it that I'm missing?

Reply:
So hypothetically if this wasn't blocked by other projects importing Veneur, there is always an advantage of preemptively bumping--or at least attempting to. The advantage is we de-risk upgrading in the future.

Example scenario: some time in the future you finally decide you want to use new Go features in Go 1.24 or something, but this repo is still on 1.18. Now you have to worry about upgrading several versions and ensuring QA _before_ you get to what you really want to do. Basically I'm just describing tech debt.

If we can prove upgrading works at every incremental stage along the way, there's much less to worry about in the future. And we get any performance benefits, opportunity to use new tooling, etc. along the way.

Reply:
The `go` line in `go.mod` that we are talking about controls the _availability_ of language features, and some behavior of the Go command (as per [docs](https://go.dev/doc/modules/gomod-ref#go-notes)).  I don't see anything in the 1.19 release notes that suggests that bumping this line actually de-risks anything.

Your PR _separately_ builds + runs the tests in this repo with Go 1.19.  That seems good to me, and ensures we behave well with the 1.19 runtime!

> hypothetically if this wasn't blocked by other projects importing Veneur

This is not hypothetical from my perspective :)

I'm OK leaving this PR open until internal Stripe repos are updated to go 1.19 if you want!  If we want to merge sooner, I think we need to keep this line at `go 1.18` (again - unless I'm missing something!)